os/pm/pm_changestate.c

```diff
@@ -221,6 +221,10 @@ int pm_changestate(enum pm_state_e newstate)
 	 */
 
 	flags = enter_critical_section();
+	/* State change only if PM is ready to state change */
+	if (!g_pmglobals.ready_to_state_change) {
```

Comment:
Let's move this into pm_idle.

Reply:
Since pm state transition can happen at various places like `Idle Thread, timer, wake handlers, etc`. Isn't it good to keep this in `pm_changestate` ?

Reply:
In the current structure, the initial start for PM is from pm_idle. It seems like a good idea to have the idle thread do nothing at all.